### PR TITLE
[NFC][hwasan] Remove code duplication in ShadowMapping::init

### PR DESCRIPTION
**Pull request by vitalybuka**
The goal to is to reorder this function to make
initialization in following order:
1. Defaults
2. Target specific overrides
3. Explicit copt<> overrides

### CODE REVIEW

File: llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp [hunk 1]
```diff
@@ -1939,10 +1943,7 @@ void HWAddressSanitizer::ShadowMapping::init(Triple &TargetTriple,
   } else if (ClWithIfunc) {
     Kind = OffsetKind::kIfunc;
     WithFrameRecord = false;
-  } else if (ClWithTls) {
```
Comment (fmayer) [inline llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp:1942]: I don't understand why we flipped this logic. before all the ifs were positive

Comment (vitalybuka) [reply llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp:1942]: ClWithTls branch is now empty - everything is in default

Comment (fmayer) [reply llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp:1942]: I understand, but standing on it's own, it's unclear why this CL makes things better.

Comment (vitalybuka) [reply llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp:1942]: It's NFC which does not need make things better, just not make things worse. The point is to instruct this shuffling to have smaler non-NFC patch to get to https://github.com/llvm/llvm-project/pull/109621/files.

**Review by fmayer — APPROVED** [review]

File: llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp (hunk 1 above)
Comment (vitalybuka) [reply llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp:1942]: Added description if it helps

Comment (fmayer) [reply llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp:1942]: yes

**Pull request merged**